### DISCUSSION TITLE
how to use langchain with bento ml

### DISCUSSION THREAD

**Discussion by Raishani** [bentoml/BentoML#4239] [Q&A] [answered]
how to use langchain with bento ml?

**Comment by parano** [comment, marked as answer]
Hi @Raishani 

For a typical LangChain app with OpenAI endpoint, you can check out https://github.com/ssheng/BentoChain as an example

If you are planning on using LangChain with Open Source LLMs, such as Llama2, see https://github.com/bentoml/OpenLLM which is built on top of BentoML, and its integration with LangChain: https://python.langchain.com/docs/integrations/llms/openllm
